sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,880 @@
+    """
+
+    train_infer_instance_count: int
+    cpu_instance_type: str
+    gpu_instance_type: str
+    graph_construction_instance_type: str
+    gsprocessing_instance_count: int
+    train_on_cpu: bool
+    volume_size_gb: int
+
+    def __post_init__(self):
+        assert (
+            self.cpu_instance_type or self.gpu_instance_type
+        ), "At least one instance type (CPU or GPU) should be specified."
+
+        if self.train_on_cpu:
+            assert (
+                self.cpu_instance_type
+            ), "Need to provide a CPU instance type when training on CPU"
+        else:
+            assert (
+                self.gpu_instance_type
+            ), "Need to provide a GPU instance type when training on GPU"
+
+        if not self.graph_construction_instance_type:
+            self.graph_construction_instance_type = self.cpu_instance_type
+            assert (
+                self.cpu_instance_type
+            ), "Need to provide a CPU instance for graph construction."
+            logging.warning(
+                "No graph processing instance type specified, using the CPU instance type: %s",
+                self.cpu_instance_type,
+            )
+
+
+@dataclass
+class TaskConfig:
+    """Pipeline/task-level configuration.
+
+    Parameters
+    ----------
+    base_job_name : str
+        Base job name for SageMaker jobs.
+    graph_name : str
+        Name of the graph.
+    input_data_s3 : str
+        S3 path to the input graph data.
+    jobs_to_run : List[str]
+        List of jobs to run in the pipeline.
+    log_level : str
+        Logging level for the jobs.
+    output_prefix : str
+        S3 prefix for the output data.
+    pipeline_name : str
+        Name for the pipeline.
+    """
+
+    base_job_name: str
+    graph_name: str
+    input_data_s3: str
+    jobs_to_run: List[str]
+    log_level: str
+    output_prefix: str
+    pipeline_name: str
+
+    def __post_init__(self):
+        # Ensure job names are valid
+        for job_type in self.jobs_to_run:
+            assert (
+                job_type in JOB_ORDER
+            ), f"Unsupported job type: '{job_type}', expected one of {list(JOB_ORDER.keys())}"
+
+        # Ensure jobs are in the right order
+        self.jobs_to_run.sort(key=lambda x: JOB_ORDER[x])
+
+        # We should only run either gconstruct or gsprocessing
+        if "gconstruct" in self.jobs_to_run and "gsprocessing" in self.jobs_to_run:
+            raise ValueError(
+                "Should not try to run both GConstruct and GSProcessing steps, "
+                f"got job sequence: {self.jobs_to_run}"
+            )
+
+        # When running gsprocessing ensure we run dist_part as well
+        if "gsprocessing" in self.jobs_to_run and "dist_part" not in self.jobs_to_run:
+            raise ValueError(
+                "When running GSProcessing need to run 'dist_part' as the following job, "
+                f"got job sequence: {self.jobs_to_run}"
+            )
```

Comment:
Maybe we want to allow pipelines that only contain a GSProcessing step?

Reply:
gconstruct is simpler. We can leave this question to the next release.